coderd/activitybump.go

```diff
@@ -70,12 +70,16 @@ func activityBumpWorkspace(ctx context.Context, log slog.Logger, db database.Sto
 		}
 
 		newDeadline := database.Now().Add(bumpAmount)
+		if !build.MaxDeadline.IsZero() && newDeadline.After(build.MaxDeadline) {
+			newDeadline = build.MaxDeadline
+		}
```

Comment:
It seems to be key part of the implementation logic. It would be cool to cover it with some unit tests.